backend/src/akvo/lumen/component/keycloak.clj

```diff
@@ -93,32 +110,147 @@
         (response {:status (:status ed)
                    :body (:reasonPhrase ed)})))))
 
+(defn tenant-admin?
+  [request-headers api-root tenant user-id]
+  (let [admin-group-id (-> (client/get (format "%s/group-by-path/akvo/lumen/%s/admin"
+                                               api-root tenant)
+                                       {:headers request-headers})
+                           :body json/decode (get "id"))
+        admins (-> (client/get (format "%s/groups/%s/members" api-root admin-group-id)
+                               {:headers request-headers})
+                   :body json/decode)
+        admin-ids (into #{}
+                        (map #(get % "id"))
+                        (filter #(and (get % "emailVerified")
+                                      (get % "enabled"))
+                                admins))]
+    (contains? (set admin-ids) user-id)))
+
+(defn fetch-user-by-id
+  "Get user by email. Returns nil if not found."
+  [request-headers api-root tenant user-id]
+  (let [resp (-> (client/get (format "%s/users/%s" api-root user-id)
+                             {:headers request-headers})
+                 :body json/decode)]
+    (assoc resp
+           "admin"
+           (tenant-admin? request-headers api-root tenant user-id))))
+
 (defn fetch-user-by-email
   "Get user by email. Returns nil if none found."
-  [request-draft api-root email]
+  [request-headers api-root email]
   (let [candidates (-> (client/get (format "%s/users?email=%s"
                                            api-root email)
-                                   request-draft)
+                                   {:headers request-headers})
                        :body json/decode)]
     ;; Since the keycloak api does a search and not a key lookup on the email
     ;; we need make sure that we have an exact match
     (first (filter (fn [candidate]
                      (= (get candidate "email") email))
                    candidates))))
 
+(defn fetch-user-groups
+  "Get the groups of the user"
+  [request-headers api-root user-id]
+  (-> (client/get (format "%s/users/%s/groups" api-root user-id)
+                  {:headers request-headers})
+      :body json/decode))
+
+(defn tenant-member?
+  "Return true for both members and admins."
+  [{api-root :api-root :as keycloak} tenant email]
```

Comment:
Not sure if it's better but usually when I don't want to rename the bindings I use `:keys` instead:

```
{:keys [api-root] :as keycloak}
```